lib/scope.js

```diff
@@ -327,47 +327,59 @@ module.exports = {
+      `input[name="${ base64_name }"][value="${ set_to }"]:not([disabled])`,  // radio
+      `*[data-saveas="${ base64_name }"] input[value="${ choice_name }"]:not([disabled])`,  // radio showif
+      `*[data-saveas="${ base64_name }"] input[value="${ set_to }"]:not([disabled])`,  // radio showif
+      `input[name="${ base64_name }"][value="True"]:not([disabled])`,  // yesno/noyes checkbox
+      `*[data-saveas="${ base64_name }"] input[value="True"]:not([disabled])`,  // yesno/noyes checkbox showif
       // Finds the group by looking for the first checkbox in the group using id, then works its way back out again
       // 'none of the above' checkbox showif???
-      await scope.page.$$(`input[name="${ name_attr_B }"]`),  // checkbox
-      await scope.page.$$(`input[name="${ name_attr_R }"]`),  // checkbox alt
-      // await scope.page.$$(`*[data-saveas="${name_attr_B}"]`),  // checkbox showif?
-      // await scope.page.$$(`*[data-saveas="${name_attr_R}"]`),  // checkbox alt showif?
+      `input[name="${ name_attr_B }"]:not([disabled])`,  // checkbox
+      `input[name="${ name_attr_R }"]:not([disabled])`,  // checkbox alt
+    ];
+
+    let handle = await scope.page.evaluateHandle(({ selectors, base64_id_of_first_choice, base64_name }) => {
+      for ( let selector of selectors ) {
+        let possible_node = document.querySelector( selector );
+        if ( possible_node ) { return possible_node; }
+      }
 
       // Put these two last so they'll come last in the loop and won't override other options
-      await scope.page.$x(`//input[@id="${ base64_id_of_first_choice }"]/ancestor::fieldset//input[contains(@class,"danota-checkbox")]`),  // 'none of the above' checkbox
-      await scope.page.$$(`:not(button)[name="${ base64_name }"]`),  // others (not showif)
-    ];
 
-    for ( let attempt of all_attempts ) {
-      // There really should be only one anyway, but they're all lists
-      if ( attempt[0] ) {
-        let tag = await attempt[0].evaluate( elem => { return elem.tagName; });
-        return { handle: attempt[0], tag: tag };
+      // 'none of the above' checkbox
+      let firstChoiceInput = document.querySelector(`input#${ base64_id_of_first_choice }`);
+      if ( firstChoiceInput ) {
+       let fieldset = firstChoiceInput.closest('fieldset');
+       let noneOfTheAboveCheckbox = fieldset.querySelector('input.danota-checkbox:not([disabled])');
+       if ( noneOfTheAboveCheckbox ) { return noneOfTheAboveCheckbox; } 
       }
-    }
 
-    // If we get out of the for loop, we didn't find the element
-    let err_message = `Cannot find the input with the variable name "${ var_name }"`;
-    if ( choice_name ) {
-      throw new ReferenceError( `${ err_message } and the value "${ choice_name }. Is it hidden?"` );
-    } else {
-      throw new ReferenceError( `${ err_message }. Is it hidden?` );
+      let other = document.querySelector(`:not(button)[name="${ base64_name }"]:not([disabled])`);
```

Comment:
Add the previously-existing `others (not showif)` above this line, so we don't lose what element this is looking for.

Reply:
Above this line would be before `firstChoiceInput`. The '`if ( choice_name )...` stuff is being deleted. Did you mean to describe putting it above `firstChoiceInput`? If so, we should discuss that more because my thinking is that this would be a catch all for anything that was left over that could possibly match the given specifiers.

Reply:
I meant: 

```javascript
      // Put these two last so they'll come last in the loop and won't override other options  <----- this talks about *two* final options 

      // 'none of the above' checkbox  <------ this is the first final optiion 
      let firstChoiceInput = document.querySelector(`input#${ base64_id_of_first_choice }`);
      if ( firstChoiceInput ) {
       let fieldset = firstChoiceInput.closest('fieldset');
       let noneOfTheAboveCheckbox = fieldset.querySelector('input.danota-checkbox:not([disabled])');
       if ( noneOfTheAboveCheckbox ) { return noneOfTheAboveCheckbox; } 
      }

      // <------ need a comment here explaining what the second of the two final options is 
      let other = document.querySelector(`:not(button)[name="${ base64_name }"]:not([disabled])`);
      if ( other ) { return other; }

      return null;
    }, { selectors, base64_id_of_first_choice, base64_name });
```

Reply:
Ah, I finally understand. Thanks!